src/korg/fit.py

```diff
@@ -0,0 +1,41 @@
+from collections.abc import Iterable
+from typing import Callable
+
+from korg._python_interface import Array1dF64, KFloat, Linelist
+
+from ._julia_import import Korgjl
+
+type KorgFitParamsType = dict[str, KFloat] | dict[str, int] | dict[str, float]
+
+
+def fit_spectrum(
+    obs_wls: Array1dF64,
+    obs_flux: Array1dF64,
+    obs_err: Array1dF64,
+    linelist: Linelist,
+    initial_guesses: KorgFitParamsType,
+    fixed_params: KorgFitParamsType = {},
```

Comment:
Most linters complain when you use a dict as a default value.
```suggestion
    fixed_params: KorgFitParamsType | None = None,
```

Reply:
done!